allauth/socialaccount/providers/battlenet/provider.py

```diff
@@ -1,18 +1,12 @@
+import re
 from allauth.socialaccount import providers
 from allauth.socialaccount.providers.base import ProviderAccount
 from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
 from allauth.socialaccount.adapter import DefaultSocialAccountAdapter
 
 
 class BattleNetSocialAccountAdapter(DefaultSocialAccountAdapter):
-    def save_user(self, request, sociallogin, form=None):
-        user = super().save_user(request, sociallogin, form)
-
-        battletag = sociallogin.account.extra_data.get("battletag")
-        if battletag:
-            user.username = battletag
-            user.save()
-        return user
+    username_regex = re.compile(r"^[\w.]+#\d+$")
```

Comment:
If BattleNet requires this particular regexp, why not simply add it to the documentation. You could even go as far doing this:

```
from allauth.account.adapter import get_adapter

if get_adapter().uername_regex != 'this-one':
  raise ImproperlyConfigured()

```

Then, we the issue remains contained within this specific adapter/provider, and developers won't be able to accidentally wrongly configure things.